src/components/AppSidebar/AppSidebarTabs.vue

```diff
@@ -213,15 +213,20 @@ export default {
 		 * Manually update the sidebar tabs according to $slots.default
 		 */
 		updateTabs() {
+			if (!this.$slots.default) {
+				this.tabs = []
+			}
+
 			// Find all valid instances of AppSidebarTab
 			const tabs = this.$slots.default.reduce((tabs, tabNode) => {
 				const tab = tabNode.componentInstance
-				if (// Find AppSidebarTab components only, take care of PascalCase and kebab-case
-					tabNode?.componentOptions?.tag.replace(/-/g, '').toLowerCase() === 'appsidebartab'
-					&& (tab?.name && typeof tab?.name === 'string')
+				// Make sure all required props are provided and valid
+				if (IsValidString(tab?.name)
 					&& IsValidString(tab?.id)
 					&& IsValidString(tab?.icon)) {
 					tabs.push(tab)
+				} else {
+					console.debug('Ignoring invalid tab', tabNode)
```

Comment:
Not sure if you want this, but the Netlify example now prints a lot of these warning, since `this.$slots.default` contains entries only consisting of whitespace:

<img width="1280" alt="Untitled" src="https://user-images.githubusercontent.com/2496460/90515939-a4232500-e163-11ea-8fd9-5acbacf2a393.png">

Reply:
Could be fixed by checking if the entry has a tag or non-whitespace text, similar to https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/AppSidebarTabs/AppSidebarTabs.vue#L230:

```this.$slots.default.filter(elem => elem.tag || elem.text.trim())```

Reply:
It only warns not when the node contains more than only whitespace.
However, it still prints a debug message on the console when you correctly provide a single node and no tabs. But since it is only a debug message it might be ok (although I would prefer to not have this output).
@skjnldsv What do you think?

Reply:
Looks like this for the empty content example (with no tabs and only a single node):

<img width="1280" alt="debug" src="https://user-images.githubusercontent.com/2496460/90684120-f3527e00-e267-11ea-9de6-37b72c63382f.png">